pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -504,20 +550,339 @@ func (r *ReconcileWildFlyServer) routeForWildFly(w *wildflyv1alpha1.WildFlyServe
+	if err != nil {
+		return false, "", fmt.Errorf("Error in probing transaction log for scaling down pod %v, error: %v", scaleDownPodName, err)
+	}
+	// Transaction log was probed, now we read the set of transactions which are in-doubt
+	jsonResult, err = wildflyutil.ExecuteMgmtOp(scaleDownPod, jbossHome, wildflyutil.MgmtOpTxnRead)
+	if err != nil {
+		return false, "", fmt.Errorf("Cannot read transactions from the transaction log for pod scaling down %v, error: %v", scaleDownPodName, err)
+	}
+	if !wildflyutil.IsMgmtOutcomeSuccesful(jsonResult) {
+		return false, "", fmt.Errorf("Cannot get list of the in-doubt transactions at pod %v for transaction scaledown", scaleDownPodName)
+	}
+	// Is the number of in-doubt transactions equal to zero?
+	transactions := jsonResult["result"]
+	txnMap, isMap := transactions.(map[string]interface{}) // typing the variable to be a map of interfaces
+	if isMap && len(txnMap) > 0 {
+		retString := fmt.Sprintf("Recovery scan to be invoked as the transaction log storage is not empty for pod scaling down pod %v, "+
+			"transaction list: %v", scaleDownPodName, txnMap)
+		return false, retString, nil
+	}
+	// Verification of the unfinished data of the WildFly transaction client (verification of the directory content)
+	lsCommand := fmt.Sprintf(`ls %s/%s/ 2> /dev/null || true`, jbossHome+standaloneServerDataDirSuffix, wftcDataDirName)
+	commandResult, err := wildflyutil.ExecRemote(scaleDownPod, lsCommand)
+	if err != nil {
+		return false, "", fmt.Errorf("Cannot query filesystem to check existing remote transactions for pod scaling down %v. "+
+			"Exec command %v", scaleDownPodName, lsCommand)
+	}
+	if commandResult != "" {
+		retString := fmt.Sprintf("WFTC data dir is not empty and scaling down of the pod '%v' will be retried."+
+			"Wildfly Transacton Client data dir path '%v', output listing: %v",
+			scaleDownPodName, jbossHome+standaloneServerDataDirSuffix+"/"+wftcDataDirName, commandResult)
+		return false, retString, nil
+	}
+	return true, "", nil
+}
+
+// matches checks if the envVar from the WildFlyServerSpec matches the same env var from the container.
+// If it does not match, it updates the container EnvVar with the fields from the WildFlyServerSpec and return false.
+func matches(container *v1.Container, envVar corev1.EnvVar) bool {
+	for index, e := range container.Env {
+		if envVar.Name == e.Name {
+			if !reflect.DeepEqual(envVar, e) {
+				container.Env[index] = envVar
+				return false
+			}
+			return true
+		}
+	}
+	//append new spec env to container's env var
+	container.Env = append(container.Env, envVar)
+	return false
+}
+
+func getWildflyServerPodStatusByName(w *wildflyv1alpha1.WildFlyServer, podName string) *wildflyv1alpha1.PodStatus {
+	for _, podStatus := range w.Status.Pods {
+		if podName == podStatus.Name {
+			return &podStatus
+		}
+	}
+	return nil
+}
+
+func containsInMap(m map[string]string, v string) bool {
```

Comment:
maybe add a comment that you're checking if a map contains the `v` string in its *values*